### PR TITLE
chore(atomix): update to Kryo 5

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Changes Kryo version from 4 to 5. Since this breaks backwards compatibility, I removed all of the fallback logic. 
The benchmark for this is `mp-spike` and I have another one for comparison `mp-baseline`. I haven't seen any performance regressions but I'll let it run for longer. I also took some profiles and the percentage of time spent serializing and deserializing seems to have decreased with this change.

 
## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [x] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by MiguelPires** [comment]
The benchmark is still running with a higher average throughput than the baseline. I didn't find any errors in the logs that point to a Kryo issue (no Kryo exceptions in 13 days its been running). Recent profiles still show that the percentage of time spent in Kryo is either lower than or similar to the baseline

**Comment by MiguelPires** [comment]
Presumably because of improvements between versions. It might also be because the partitions were slightly better distributed among the nodes in one benchmark than the other one. The difference in the 10 day average is also going down so it might go to zero

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/network/LargeMessageSizeTest.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 public final class LargeMessageSizeTest {
 
-  private static final DataSize MAX_MESSAGE_SIZE = DataSize.ofMegabytes(4);
+  private static final DataSize MAX_MESSAGE_SIZE = DataSize.ofMegabytes(5);
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/network/LargeMessageSizeTest.java:27]: Why do you need this change?

File: atomix/utils/src/main/java/io/atomix/utils/serializer/CompatibleKryoPool.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    this.friendlyName = friendlyName;
+    this.classLoader = classLoader;
+    this.registeredBlocks = registeredBlocks;
+  }
+
+  @Override
+  protected Kryo create() {
+    final Kryo kryo = new Kryo();
+    final CompatibleFieldSerializerConfig serializerConfig = new CompatibleFieldSerializerConfig();
+    serializerConfig.setChunkedEncoding(true);
```
Comment (deepthidevaki) [inline atomix/utils/src/main/java/io/atomix/utils/serializer/CompatibleKryoPool.java:57]: May be add a small doc here about these configurations?

**Review by deepthidevaki — COMMENTED** [review]
Looks good :+1: Just some minor comments.

Since we are already making changes here, how about we assign ids to the registered classes. Currently it is automatically assigned, which means if we remove or add new classes it will break the compatibility. It is out of scope for this PR, but we could think about it as a follow up. wdyt?

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/network/LargeMessageSizeTest.java (hunk 1 above)
Comment (MiguelPires) [reply qa/integration-tests/src/test/java/io/zeebe/broker/it/network/LargeMessageSizeTest.java:27]: The encoded size is larger now so the size of the entry exceeded 5Mb

**Comment by MiguelPires** [comment]
I think assigning the ids manually might be a bit error prone, especially since some namespaces include another namespace's registrations. If we want to avoid the `register(Void.class)`, we can just add a method to skip X ids and call that when we remove classes.

**Comment by MiguelPires** [comment]
The 1 partition benchmarks I started yesterday don't show any changes in performance btw (similar avg throughput and execution latency. Buffer pool is a bit higher maybe due to the larger serialization output)

**Review by deepthidevaki — APPROVED** [review]
Thanks @MiguelPires :+1:

**Comment by MiguelPires** [comment]
Thanks for the review @deepthidevaki 

bors r+

**Pull request merged**